django/contrib/auth/tokens.py

```diff
@@ -128,5 +135,40 @@ def _now(self):
         # Used for mocking in tests
         return datetime.now()
 
+    def _xor_encrypt_decrypt(self, uid):
```

Comment:
The two "public" APIs being added to this method should be fully tested, including the corner cases described in the docstring (thanks for the thorough details).